runtime/dart_isolate.cc

```diff
@@ -130,23 +188,12 @@ DartIsolate::DartIsolate(const Settings& settings,
                   settings.log_tag,
                   settings.unhandled_exception_callback,
                   DartVMRef::GetIsolateNameServer()),
-      settings_(settings),
-      isolate_snapshot_(std::move(isolate_snapshot)),
-      child_isolate_preparer_(std::move(child_isolate_preparer)),
-      isolate_create_callback_(isolate_create_callback),
-      isolate_shutdown_callback_(isolate_shutdown_callback),
-      is_root_isolate_(is_root_isolate),
-      is_group_root_isolate_(is_group_root_isolate) {
-  FML_DCHECK(isolate_snapshot_) << "Must contain a valid isolate snapshot.";
+      is_root_isolate_(is_root_isolate) {
   phase_ = Phase::Uninitialized;
 }
 
 DartIsolate::~DartIsolate() = default;
```

Comment:
Can we add an assertion here that checks that if there is a message handling task runner set, the destructor is on its thread?